src/libprojectM/Renderer/MenuText.cpp

```diff
@@ -0,0 +1,163 @@
+#ifdef USE_TEXT_MENU
+
+#include "MenuText.h"
+
+#include "Common.hpp"
+
+#define GLT_IMPLEMENTATION
+#include "gltext.h"
+
+MenuText::MenuText(int viewportWidth)
+    : _viewportWidth(viewportWidth)
+{
+    gltInit();
+    _glTextInstance = gltCreateText();
+}
+
+MenuText::~MenuText()
+{
+    if (_glTextInstance)
+    {
+        gltDeleteText(_glTextInstance);
+        _glTextInstance = nullptr;
+    }
+}
+
+void MenuText::SetViewportWidth(int viewportWidth)
+{
+    _viewportWidth = viewportWidth;
+}
+
+void MenuText::CleanUp()
+{
+    gltTerminate();
+}
+
+void MenuText::DrawBegin() const
```

Comment:
Moved these calls outside Draw() as binding/unbinding shaders and textures also adds some serious overhead when drawing the preset menu.